awx/main/analytics/core.py

```diff
@@ -324,21 +324,25 @@ def gather(dest=None, module=None, subset=None, since=None, until=None, collecti
                     settings.AUTOMATION_ANALYTICS_LAST_ENTRIES = json.dumps(last_entries, cls=DjangoJSONEncoder)
 
         if collection_type != 'dry-run':
+            # Clean up temp files regardless of success or failure
+            for fpath in tarfiles:
+                if os.path.exists(fpath):
+                    os.remove(fpath)
+
+            # Update timestamps only on success
```

Comment:
Is this change intentional?:

* previously `AUTOMATION_ANALYTICS_LAST_GATHER` was updated *outside* of that `if succeeded`:
* line 337 still says `AUTOMATION_ANALYTICS_LAST_GATHER` is set whether collection succeeds or fails`

So it was documented as intentionally updating in both cases, and it actually did that. Should we be changing this here?

Reply:
@himdel Yes, from what I understand, this is exactly the point that is causing the issue. And the documentation from line 337 onwards no longer makes sense if this change is valid and I intend to remove it soon.

Reply:
Just to be clear, I'm not commenting on the move of `os.remove` outside success-only,
but on the move of the timestamp update inside success-only.

I'm pretty sure the first fixes the bug,
but the second should be unrelated?

Reply:
@himdel is correct, this needs to stay the same 
```
settings.AUTOMATION_ANALYTICS_LAST_GATHER = until 
```

must always update (not be based on succeeded )

(otherwise we'd see this job running every few minutes instead of the set interval)

Reply:
seems like the "succeeded" condition can be removed entirely then